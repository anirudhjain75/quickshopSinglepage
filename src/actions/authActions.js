import firebase from 'firebase';

export const emailChange = (text) => {
    return {
        type: 'EMAIL_CHANGED',
        payload: text
    };
};

export const passChange = (text) => {
    return {
        type: 'PASSWORD_CHANGED',
        payload: text
    };
};

export const FirstNameChange = (text) => {
    return {
        type: 'FIRST_NAME_CHANGED',
        payload: text
    }
};

export const LastNameChange = (text) => {
    return {
        type: 'LAST_NAME_CHANGED',
        payload: text
    }
};

export const loginUser = ({email, password}) => {
    return (dispatch) => {
        dispatch({ type: 'LOGIN_USER' });

        firebase.auth().signInWithEmailAndPassword(email, password)
            .then(user => loginUserSuccess(dispatch, user))
            .catch((error) => {
                console.log(error);
                firebase.auth().createUserWithEmailAndPassword(email, password)
                    .then(user => loginUserSuccess(dispatch, user))
                    .catch(() => loginUserFail(dispatch))
            })
    };
};

const loginUserFail = (dispatch) => {
    dispatch({
        type: 'LOGIN_USER_FAIL'
    });
};

const loginUserSuccess = (dispatch, user) => {
    dispatch({
        type: 'LOGIN_USER_SUCCESS',
        payload: user
    });
};
