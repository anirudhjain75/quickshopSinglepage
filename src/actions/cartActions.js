var uuid = require('uuid/v1')

export const addToCart = (item) => {
    return {
        type: 'ADD_TO_CART',
        payload: {...item, uuid: uuid()}
    }
};

export const removeFromCart = (uuid) => {
    return {
        type: "REMOVE_FROM_CART",
        payload: uuid
    }
};