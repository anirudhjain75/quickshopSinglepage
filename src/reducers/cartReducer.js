const initialState = {
    cart: []
};

export default (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_TO_CART' :
        {
            state.cart.map((item) => {
                if(action.payload.name === item.name)
                {
                    item.quantity = item.quantity + action.payload.quantity;
                }
            });
            return {
                ...state,
                cart: [ ...state.cart, action.payload]
            };
        }
        case 'REMOVE_FROM_CART' : {
            let temp = state.cart;
            const index = temp.indexOf(action.payload);
            temp.splice(index, 1);
            return {
                ...state,
                cart: temp
            };
        }
        default:
            return state;
    }
}