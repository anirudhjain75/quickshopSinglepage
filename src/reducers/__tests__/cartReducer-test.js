import cartReducer from '../cartReducer';
import {addToCart, removeFromCart} from '../../actions';

test("add to cart", ()=>{
    const state = {
        cart: []
    };
    const data = {
        name: 'Apple 1KG',
        price: 100
    };
    expect(cartReducer(state, addToCart(data))).toMatchSnapshot();
});

test("remove from cart", () => {
   const state = {
       cart: [{
           name: "Apple 1KG",
           price: 100
       }]
   };
   const data = {
       name: "Apple 1KG",
       price: 100
   };
   expect(cartReducer(state, removeFromCart(data))).toMatchSnapshot();
});

test("default case", () => {
    const state = {
        cart: []
    };
    const data = {
        name: "Apple 1KG",
        price: 100
    };
    expect(cartReducer(state, {type: "default", payload: data})).toMatchSnapshot();
});