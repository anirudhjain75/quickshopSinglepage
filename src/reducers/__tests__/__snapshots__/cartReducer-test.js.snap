// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add to cart 1`] = `
Object {
  "cart": Array [
    Object {
      "name": "Apple 1KG",
      "price": 100,
    },
  ],
}
`;

exports[`default case 1`] = `
Object {
  "cart": Array [],
}
`;

exports[`remove from cart 1`] = `
Object {
  "cart": Array [],
}
`;
